weave/integrations/langchain/langchain_test.py

```diff
@@ -0,0 +1,494 @@
+    assert len(calls) == 8
+    flattened = flatten_calls(calls)
+
+    got = [(op_name_from_ref(c.op_name), d, c.parent_id) for (c, d) in flattened]
+    ids = [c.id for (c, _) in flattened]
+
+    exp = [
+        ("langchain.Chain.RunnableSequence", 0, None),
+        ("langchain.Prompt.PromptTemplate", 1, ids[0]),
+        ("langchain.Llm.ChatOpenAI", 1, ids[0]),
+        ("openai.chat.completions.create", 2, ids[2]),
+        ("langchain.Chain.RunnableSequence", 0, None),
+        ("langchain.Prompt.PromptTemplate", 1, ids[4]),
+        ("langchain.Llm.ChatOpenAI", 1, ids[4]),
+        ("openai.chat.completions.create", 2, ids[6]),
+    ]
+    assert got == exp
+
+
+@pytest.mark.skip_clickhouse_client
+@pytest.mark.vcr(
+    filter_headers=["authorization"],
+    allowed_hosts=["api.wandb.ai", "localhost", "trace.wandb.ai"],
+    before_record_request=filter_body,
+)
+def test_simple_chain_batch(
+    client: WeaveClient,
+) -> None:
+    from langchain_core.prompts import PromptTemplate
+    from langchain_openai import ChatOpenAI
+
+    api_key = os.environ.get("OPENAI_API_KEY", "sk-1234567890abcdef1234567890abcdef")
+
+    llm = ChatOpenAI(
+        model_name="gpt-3.5-turbo", openai_api_key=api_key, temperature=0.0
+    )
+    prompt = PromptTemplate.from_template("1 + {number} = ")
+
+    llm_chain = prompt | llm
+    _ = llm_chain.batch([{"number": 2}, {"number": 3}])
+
+    res = client.server.calls_query(tsi.CallsQueryReq(project_id=client._project_id()))
+    assert_correct_calls_for_chain_batch(res.calls)
+
+
+def assert_correct_calls_for_chain_batch_from_op(calls: list[tsi.CallSchema]) -> None:
+    assert len(calls) == 9
+    flattened = flatten_calls(calls)
+
+    got = [(op_name_from_ref(c.op_name), d, c.parent_id) for (c, d) in flattened]
+    ids = [c.id for (c, _) in flattened]
+
+    exp = [
+        ("run_batch", 0, None),
+        ("langchain.Chain.RunnableSequence", 1, ids[0]),
+        ("langchain.Prompt.PromptTemplate", 2, ids[1]),
+        ("langchain.Llm.ChatOpenAI", 2, ids[1]),
+        ("openai.chat.completions.create", 3, ids[3]),
+        ("langchain.Chain.RunnableSequence", 1, ids[0]),
+        ("langchain.Prompt.PromptTemplate", 2, ids[5]),
+        ("langchain.Llm.ChatOpenAI", 2, ids[5]),
+        ("openai.chat.completions.create", 3, ids[7]),
+    ]
+    assert got == exp
+
+
+@pytest.mark.skip_clickhouse_client
+@pytest.mark.vcr(
+    filter_headers=["authorization"],
+    allowed_hosts=["api.wandb.ai", "localhost", "trace.wandb.ai"],
+    before_record_request=filter_body,
+)
+def test_simple_chain_batch_inside_op(
+    client: WeaveClient,
+) -> None:
+    # This test is the same as test_simple_chain_batch, but ensures things work when nested in an op
+    from langchain_core.prompts import PromptTemplate
+    from langchain_openai import ChatOpenAI
+
+    api_key = os.environ.get("OPENAI_API_KEY", "sk-1234567890abcdef1234567890abcdef")
```

Comment:
api key

Reply:
Haha,  :laughing: this is a dummy key made to look the same cosmetically